sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -652,10 +652,12 @@ object DataSourceStrategy {
  */
 object PushableColumn {
   def unapply(e: Expression): Option[String] = {
-    def helper(e: Expression) = e match {
-      case a: Attribute => Some(a.name)
+    import org.apache.spark.sql.connector.catalog.CatalogV2Implicits.MultipartIdentifierHelper
+    def helper(e: Expression): Option[Seq[String]] = e match {
+      case a: Attribute => Some(Seq(a.name))
+      case s: GetStructField => helper(s.child).map(_ :+ s.childSchema(s.ordinal).name)
```

Comment:
shall we add a legacy config to skip it? Then if we break some sources, there is a way to workaroud.

Reply:
Do you mean the nested predicate pushdown? It will be easy for this, but for `dots`, it's harder since if we don't add it here, we have to add it before calling the ORC filter API.

Reply:
I'd expect something like
```
if (legecyConfOn) {
  e match {
    case a: Attribute => Some(a.name)
    case _ => None
} else {
  def help ....  // the new code.
}
```

Reply:
Parquet side doesn't need to handle the legacy config. If users turn on the legacy config and the column name has dot, we just ignore that filter, which affects performance but not a correctness issue.